lib/get-transitive-data.js

```diff
@@ -1,35 +1,53 @@
-/** Create transitive data for journeys from a particular origin to a particular destination */
-
+// @flow
 import lonlat from '@conveyal/lonlat'
 import {hsl} from 'd3-color'
-import slice from 'lodash/slice'
 
+import {
+  MAX_DISSIMILARITY,
+  MAX_TRANSITIVE_PATHS
+} from './constants'
 import {pixelToLat, pixelToLon} from './mercator'
-import {ITERATION_WIDTH} from './origin'
-
-// maximum path segments before we conclude that either R5 or Browsochrones is off its rocker and bail
-// (we do this so that we don't generate and attempt to render crazy paths; it implies a bug but it's better to log
-//  an erro message than crash the user's browser)
-const MAX_PATH_SEGMENTS = 8
-
-// arbitrary value that determines how aggressively to cluster results (higher is more aggressive, and a high number is something like 1e-3)
-const MAXIMUM_DISSIMILARITY = 5e-6
+import {getStopPatternStopSets} from './origin'
+
+import type {
+  Journey,
+  Origin,
+  PathDescriptor,
+  Pattern,
+  Place,
+  Point,
+  Query,
+  Segment,
+  Stop,
+  StopPatternStops
+} from './types'
 
+/**
+ * Create transitive data for journeys from a particular origin to a
+ * particular destination
+ */
 export default function getTransitiveData ({
   from,
   log,
   origin,
   paths,
   query,
-  stopTreeCache,
   to,
   transitiveNetwork
-}) {
-  // create the journeys
-  const output = {
-    places: [],
-    journeys: []
-  }
+}: {
+  from: Point,
+  log: Function,
+  origin: Origin,
+  paths: Array<PathDescriptor>,
+  query: Query,
+  to: Point,
+  transitiveNetwork: any
```

Comment:
Should we at least define a TransitiveNetwork type that has all the top level members (journeys, etc) as type any so it at least prevents you from accidentally passing in totally the wrong type of object?